jetstream/tasks.py

```diff
@@ -137,35 +132,67 @@ def send_reports():
-
-            payload_ubr = {"allocation_source_id":source.source_id, "username":user.username, "burn_rate":burn_rate, "compute_used":compute_used}
-            EventTable.create_event("user_allocation_snapshot_changed", payload_ubr, user.username)
-        compute_used_total = allocation_source_total_compute.get(source.name,0)
-        global_burn_rate = allocation_source_total_burn_rate.get(source.name,0)
-        if compute_used_total != 0:
-            logger.info("Total usage for AllocationSource %s (%s-%s) = %s (Burn Rate: %s)" % (source.name, start_date, end_date, compute_used_total, global_burn_rate))
-        payload_as = { 
-            "allocation_source_id":source.source_id, 
-            "compute_used":compute_used_total,
-            "global_burn_rate":global_burn_rate
-        }
-        EventTable.create_event("allocation_source_snapshot", payload_as,source.name)
+    end_date = end_date or timezone.now()
+    # TODO: Read this start_date from last 'reset event' for each allocation source
+    start_date = start_date or '2016-09-01 00:00:00.0-05'
+    all_data = create_report(start_date, end_date)
+
+    user_allocation_snapshots = {}
+    unique_usernames = set()
+
+    for row in all_data:
+        key = (row['allocation_source'], row['username'])
+        compute_used, burn_rate = user_allocation_snapshots.get(key, (0.0, 0))
+        new_compute_used = compute_used + float(row['applicable_duration'])
+        new_burn_rate = int(row['burn_rate'])
+        user_allocation_snapshots[key] = (new_compute_used, new_burn_rate)
+
+        unique_usernames.add(row['username'])
+
+    allocation_source_ids = {obj['name']: obj['id'] for obj in AllocationSource.objects.all().values('name', 'id')}
+    relevant_users = {obj['username']: obj['id'] for obj in
+                      AtmosphereUser.objects.filter(username__in=unique_usernames).values(
+                          'username', 'id')}
+
+    allocation_source_burn_rates = collections.Counter()
+    for key, snapshot_numbers in user_allocation_snapshots.iteritems():
+        allocation_source_name, username = key
+        compute_used, burn_rate = snapshot_numbers
+        try:
+            allocation_source_id = allocation_source_ids[allocation_source_name]
+        except KeyError:
+            # This allocation source does not exist in our database yet. Create it? Skip for now. Could be 'N/A' as well
+            continue
+        snapshot, created = UserAllocationSnapshot.objects.update_or_create(
+            allocation_source_id=allocation_source_id,
+            user_id=relevant_users[username],
+            defaults={
+                'compute_used': round(compute_used / 3600, 2),
+                'burn_rate': burn_rate
+            }
+        )
+        allocation_source_burn_rates[allocation_source_name] += burn_rate
+
+    tas_api_obj = TASAPIDriver()
```

Comment:
`update_snapshot()` takes a start_date/end_date args that are not used in the code below this line..

It might be best if we separate the concerns here by keeping this code in a different function.